front/app/components/AssignBudgetControl/index.tsx

```diff
@@ -157,20 +160,30 @@ const AssignBudgetControl = memo(
       };
 
       setProcessing(true);
-
       if (!isNilOrError(basket)) {
         const basketIdeaIds = basket.relationships.ideas.data.map(
           (idea) => idea.id
         );
         const isInBasket = basketIdeaIds.includes(ideaId);
-
+        let isPermitted = true;
         let newIdeas: string[] = [];
 
         if (isInBasket) {
           newIdeas = basket.relationships.ideas.data
             .filter((basketIdea) => basketIdea.id !== idea.data.id)
             .map((basketIdea) => basketIdea.id);
         } else {
+          // If new idea causes exceeded budget, emit an error
+          if (
+            basketTotal &&
+            maxBudget &&
+            ideaBudget &&
+            basketTotal + ideaBudget > maxBudget
+          ) {
+            eventEmitter.emit(BUDGET_EXCEEDED_ERROR_EVENT);
+            isPermitted = false;
```

Comment:
Emitting the error seemed like the best solution to me since we need it in the BudgetingCTABar. I couldn't see any way to access it in that component otherwise :thinking: